Feat/drawer init close on mobile

kosmasntokos opened the pull request:
Hide the drawer sidebar on mobile, also when resizing the browser window.

kosmasntokos commented on src/components/drawerBar.js:
```diff
@@ -31,6 +30,34 @@
+    // End window resize 
+
+    // Check if mobile width
+    if(windowDimensions.width <= 600){
```
We chose 600px because this is also done in the css. Take a look at -> [`@media ${mediaMinWidth(600)}`]

dylanBetty commented:
Hey @kosmasntokos! Taking a second look at this I would suggest to simplify the resize implementation slightly. I think something like this would be all you need for it to work:
```javascript
    const [windowWidth, setWindowWidth] = useState(window.innerWidth);
    const [open, setOpen] = useState(isOpen);
    const maxMobileWidth = 600px;

    if (windowWidth <= maxMobileWidth && open) {
      setOpen(false);
    }
    if (windowWidth >= maxMobileWidth + 1 && !open) {
      setOpen(true);
    }

    useEffect(() => {
      const setWidth = () => {
        setWindowWidth(window.innerWidth);
      };
      window.addEventListener('resize', setWidth);
    }, []);

```
Adding the `&& open` checks also prevents the 'too many re-renders' error.

The components in the render method below would then also have to make use of the new `open` state.
```javascript
 <Drawer open={open} /> // instead of <Drawer open={isOpen} />
```

Let me know if you have any questions about it. :ok_hand:

kosmasntokos commented:
> 

Hi @dylanBetty,

The code you sent me looks clean. However, it still doesn't work properly. The drawer is closed neatly on mobile, but when I click on the button the drawer doesn't appear anymore. This is because there is a step missing logic.

I added the following code (the OR in the second If statement)
`    if (windowWidth <= maxMobileWidth && open) {
      setOpen(false);
    }

    if (windowWidth >= maxMobileWidth + 1 && !open || windowWidth <= maxMobileWidth && !open) {
      setOpen(true);
    }
`
, but then I immediately get the "too many re-renders" error. Do you know why that happens? I am not an expert in JSX.

I've now changed my code with your suggestions, but still with the same check on open. Would you like to see how I can still use setOpen?

`
    let open = isOpen;
    const [windowWidth, setWindowWidth] = useState(window.innerWidth);
    const maxMobileWidth = 600;

    useEffect(() => {
      const setWidth = () => {
        setWindowWidth(window.innerWidth);
      };
      window.addEventListener('resize', setWidth);
    }, []);

    if (windowWidth <= maxMobileWidth) {
      open = !isOpen;
    }
`

Thank you in advance!

sventruschel commented:
Verified with Dylan, will close this PR if these changes are still necessary a new PR can be created
